### DISCUSSION TITLE
Character generation: How does this work and affect the model?

### DISCUSSION THREAD

**Discussion by Woisek** [oobabooga/text-generation-webui#2683] [Q&A]
So, I'm starting out to get an understanding of all this. And since I've got ooba finally get to work, I want to dive into this.
I understand that there are different models available, roughly three categories, as I understood it:
chat, instructions and role play.
So I want to start with the role play models. I found out, that "pygmalion" models are suited for that. I found out, that a "character" can/should be created to use it. And here is my first hole I have to fill. With the options that are should be filled, what exactly should one fill in?
How does this work? I see fields for names and greetings, that's clear. I also see a field for "context" and that's what I need to know, what to fill in there. In what manner, what form, or generally speaking: What does this context do? Is this used to "shape" the character? 🤔
I haven't found anything online where I can study "finished" characters, so that I get an idea of it, so I have to ask here and hope that someone can enlighten me, please. 🙂
Thanks!

**Comment by BetaDoggo** [comment]
For examples there's a [booru plus](https://booru.plus/+pygmalion?c=azbgfrzaxwmt2&o=top#p279a241)(Lots of NSFW so be prepared) page for Pygmalion cards. Some of them are for other models/platforms but they should be labeled as such. The cards are pngs with their prompts embedded into them. You can view the prompts by importing them into the ui while in chat mode.

**Reply by Woisek** [reply]
OK, thanks. Anything else I could look up? Like, a syntax or so?
I saw different types of how this is filled in. Some use just plain text with YOU and <CHARACTERNAME> and others have some kind of ... JSON type text, like
'[character("Angel Devil")\n{\nspecies("Devil" + "Angel Devil")\nmind("lazy" + "pacifist" + "socially withdrawn" + "secretly kind" + "secretly a pervert")\npersonality("lazy" + "pacifist" + "socially withdrawn" + [...]
Is this even valid? Or for a different kind of text generation? 🤔
Thanks again for this first hint.

Edit:
I don't know I if I understood correctly, but did you mean dragging the image into the place where it says "TavernAI PNG format"? 🤔

**Reply by BetaDoggo** [reply]
I believe both formatting styles should work with pygmalion but plain text formatting is easier to write and more universal so some people prefer it. 

Here are some guides I found in the semiofficial Pygmalion discord for the formatting:
https://rentry.co/plists_alichat_avakson
https://rentry.co/alichat
https://rentry.co/cringeguide
https://rentry.co/WPP_For_Dummies


Yes dragging the image into the TavernAI PNG format box under the upload tab should import the character automatically.

**Reply by ossirytk** [reply]
You can also use yaml format. There is an example character in the repo in the characters folder. Yaml is basically as readable as plain text and the webui supports it. You can also convert json files to yaml files for easier readability. Results may wary since some character creators refuse to use linbreaks. No idea if this is some compatibility issue or they are min/maxing token count, but it kinda makes for a tough read.

**Reply by Woisek** [reply]
Many thanks! In the meantime I also found Aitrepreneurs video on that topic -> https://www.youtube.com/watch?v=jhLHa9-JwDM and it was very helpful.
I will look up the links, this is really a fascinating field. 😀 Thanks again!
